### PR TITLE
[cxx-interop] Lower macOS deployment target version for the Cxx module

### PR DESCRIPTION
**Pull request by egorzhdan**
The Cxx module does not rely on any modern OS features. It can be deployed to older macOS versions.

rdar://117699474

### CODE REVIEW

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@egorzhdan what about the CxxStdlib module?

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please test Windows

**Review by hyp — APPROVED** [review]

**Comment by hyp** [comment]
@egorzhdan please pick this to 5.10 as well.

**Review by etcwilde — APPROVED** [review]
Mechanically, yeah, I think this drills all of the appropriate holes to plumb through a different deployment target. Will we need to be able to alter the deployment target on other platforms too, or is this guaranteed to only be a macOS thing?

File: stdlib/cmake/modules/SwiftSource.cmake [hunk 1]
```diff
@@ -236,6 +237,10 @@ function(_add_target_variant_swift_compile_flags
 
   if("${sdk}" IN_LIST SWIFT_DARWIN_PLATFORMS)
     set(sdk_deployment_version "${SWIFT_SDK_${sdk}_DEPLOYMENT_VERSION}")
+    if("${sdk}" STREQUAL "OSX" AND DEFINED VARIANT_DEPLOYMENT_VERSION_OSX)
```
[the remark this answers is not in the document]
Comment (egorzhdan) [reply stdlib/cmake/modules/SwiftSource.cmake:240]: We might potentially need this on other Darwin platforms in the future, but for now we only expect to use this for macOS. I can add the corresponding CMake logic for iOS, etc. if you think it's worth having it in tree.

**Pull request merged**